doc/adrs/161_add_support_for_actions_in_tree_item_menu.adoc

```diff
@@ -0,0 +1,182 @@
+= ADR-161 Add support for actions in tree item menu
+
+== Context
+
+Currently, actions in tree item context menu are hard coded in the front end and one cannot add or remove an action without recompiling the front end code.
+
+== Decision
+
+We need to provide a general way to handle actions in tree item contextual menu that does not require front end code modification.
+
+=== GraphQL
+
+First we need to define a new query that returns the list of tree item contextual menu action.
+
+```graphql
+type TreeDescription implements RepresentationDescription {
+  ...
+  contextMenu(treeItemId: ID!): [TreeItemContextActionMetadata!]!
+  ...
+}
+
+interface TreeItemContextActionMetadata {
+  id: ID!
+  label: String!
+  iconURL: [String!]!
+}
+
+type TreeItemSimpleContextActionMetadata implements TreeItemContextActionMetadata {
+  id: ID!
+  label: String!
+  iconURL: [String!]!
+}
+
+type TreeItemFetchContextActionMetadata implements TreeItemContextActionMetadata {
+  id: ID!
+  label: String!
+  iconURL: [String!]!
+  urlToFetch : String!
+  fetchKind : FetchKind!
+}
+
+enum FetchKind {
+  DOWNLOAD
+  OPEN
+}
+```
+
+For executing an action, we need to add a mutation to call the selected action when it is a simple action. 
+
+```graphql
+extend type Mutation {
+  ...
+  invokeTreeItemContextAction(input: InvokeTreeItemSimpleContextActionInput!): InvokeTreeItemSimpleContextActionPayload!
+  ...
+}
+
+type InvokeTreeItemSimpleContextActionInput {
+  treeId
```

Comment:
each mutation has is own id
You will nedd the `editingContextId`
Prefer `representationId` rather than `treeId`

Reply:
Done